kubeadm: Use go:embed for readability

sivchari opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes https://github.com/kubernetes/kubeadm/issues/3200

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

neolit123 reviewed (commented):
thanks for the pr. does this cover all places where we embed test input in kubeadm tests? i think there are more, so better to change all for consistency.

sivchari commented:
@neolit123 
Thanks for your comment.
As I wrote in the issue on https://github.com/kubernetes/kubeadm/issues/3200 this PR is still WIP. There are many parts to be addressed, and I created the PR first to share the image without applying all of them to see if we can proceed with this fix.

neolit123 commented:
/kind cleanup
/release-note-none

HirazawaUi reviewed (commented):
Thank you for opening this PR; it will be helpful for us. However, I have a small suggestion:

Could we use more semantic file names, such as: `user1.yaml -> config_out_k8s_user1.yaml`?

neolit123 reviewed (commented):
@sivchari any plans to continue the work here?
the change was overall a net positive.

sivchari commented:
@neolit123 
Sorry for the late response. I'm willing to continue to tackle it.

sivchari commented:
/reopen

Sorry for much delayed. There were some troubles with my life. These were already settled, so I'd continue to work on it.

sivchari commented:
/reopen

sivchari commented:
Hi, @neolit123 @HirazawaUi 
PTAL, if you have time.

neolit123 reviewed (commented):
just one comment.
thanks for continuing the work on this.
like i said originally, i like this refactor.

/approve
/hold

sivchari committed:
kubeadm: Use go:embed for readability

Migrate test fixtures to use go:embed for improved readability
and maintainability across kubeadm test files.

Signed-off-by: sivchari <shibuuuu5@gmail.com>

neolit123 reviewed (commented):
LGTM
@HirazawaUi  PTAL as well if you have some time

HirazawaUi reviewed (commented):
/lgtm
/hold cancel
